API/projectClass/backgroundsClass.m

```diff
@@ -121,11 +138,11 @@
             obj.backgrounds.removeRow(row);
         end
         
-        function obj = setBackground(obj, row, varargin)
+        function obj = setBackground(obj, row, allowedNames, varargin)
```

Comment:
Is it possible to refactor so the setters are called after all the validation is done? This way if any validation fails you have not changed half of the parameters already. 
```
% All the Validation(
...
 obj.backgrounds.setValue(row, 1, inputBlock.name);           
 obj.backgrounds.setValue(row, 2, inputBlock.type);
 obj.backgrounds.setValue(row, i + 3, value);
```